File: CustomSpinner/app/src/main/java/com/example/customspinner/CountryAdapter.java
package com.example.customspinner;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

public class CountryAdapter extends BaseAdapter
{
    String countryar[];
    int imgar[];
    Context context;

    CountryAdapter(String[] countryar,int imgar[],Context context)
    {
        this.countryar=countryar;
        this.imgar=imgar;
        this.context=context;
    }

    @Override
    public int getCount() {
        return countryar.length;
    }

    @Override
    public Object getItem(int i) {
        return null;
    }

    @Override
    public long getItemId(int i) {
        return 0;
    }

    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {
        View v= LayoutInflater.from(context).inflate(R.layout.imagespinner,null);
        TextView txtcountryname = v.findViewById(R.id.txtcountryname);
        ImageView imgCountryflag = v.findViewById(R.id.imgCountryflag);
        txtcountryname.setText(countryar[i]);

        imgCountryflag.setImageResource(imgar[i]);
        return v;
    }
}

File: Recycleviewinvoke/app/src/main/java/com/example/recycleviewinvoke/CountryAdapter.kt
package com.example.recycleviewinvoke

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class CountryAdapter(val list : ArrayList<CountryModal>,var d:((Int,String) -> Unit)) : RecyclerView.Adapter<CountryAdapter.MyViewHolder>() {
    class MyViewHolder(view : View) : RecyclerView.ViewHolder(view) {

        var txtcode : TextView = view.findViewById(R.id.txtcode)
        var txtcountry : TextView = view.findViewById(R.id.txtcountry)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int):MyViewHolder {
        var view= LayoutInflater.from(parent.context).inflate(R.layout.data_display,parent,false)
        var myViewHolder = MyViewHolder(view)
        return myViewHolder
    }

    override fun getItemCount(): Int {
        return list.size

    }

    override fun onBindViewHolder(holder:MyViewHolder, position: Int) {
        holder.txtcode.setText(list[position].countrycode.toString())
        holder.txtcountry.setText(list[position].name.toString())

    }


}